DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -624,34 +624,72 @@ public async Task<ThreadQueryResult> ListJoinedPrivateArchivedThreadsAsync(DateT
     }
 
     /// <summary>
-    /// Deletes multiple messages if they are less than 14 days old.  If they are older, none of the messages will be deleted and you will receive a <see cref="BadRequestException"/> error.
+    /// Deletes multiple messages if they are less than 14 days old.  If they are older, none of the messages will be deleted and you will receive a <see cref="ArgumentException"/> error.
     /// </summary>
     /// <param name="messages">A collection of messages to delete.</param>
     /// <param name="reason">Reason for audit logs.</param>
-    /// <returns></returns>
-    /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ManageMessages"/> permission.</exception>
-    /// <exception cref="NotFoundException">Thrown when the channel does not exist.</exception>
-    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
-    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public async Task DeleteMessagesAsync(IEnumerable<DiscordMessage> messages, string reason = null)
+    /// <remarks>One api call per 100 requests</remarks>
+    public async Task<int> DeleteMessagesAsync(IReadOnlyList<DiscordMessage> messages, string reason = null)
```

Comment:
Should be `IEnumerable` as it's more widely supported and there are no `IReadOnlyList` specific properties or methods used.

Reply:
https://discord.com/channels/379378609942560770/379386901725052928/1189188309788917800

Reply:
That comment was specific to the `ToArray` call and can be discarded due to our new logic